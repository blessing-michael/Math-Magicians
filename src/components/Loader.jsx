import React from 'react';
import { loader } from '../assets';
import '../styles/Loader.css';

const Loader = () => (
  <div className="loader">
    <img src={loader} alt="loader" className="loader-img" />
    <p className="loader-tag">
      We are getting you a quote
      <br />
      Please wait...
    </p>
  </div>
);

export default Loader;
