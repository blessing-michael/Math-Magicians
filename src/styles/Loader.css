.loader {
    z-index: 10;
    background: rgba(0, 0, 0, 0.7);
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
  }
  
  .loader-img {
    width: 100px;
    height: 100px;
    object-fit: contain;
  }
  
  .loader-tag {
    font-family:  sans-serif;
    align-self: center;
    margin-top: 20px;
    font-weight: 700;
    color: #fff;
  }