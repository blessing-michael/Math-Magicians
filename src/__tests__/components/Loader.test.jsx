import React from 'react';
import renderer from 'react-test-renderer';
import Loader from '../../components/Loader';

test('renders Loader correctly', () => {
  const component = renderer.create(<Loader />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
