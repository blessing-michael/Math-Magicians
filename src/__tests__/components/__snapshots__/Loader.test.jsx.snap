// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Loader correctly 1`] = `
<div
  className="loader"
>
  <img
    alt="loader"
    className="loader-img"
    src="loader.svg"
  />
  <p
    className="loader-tag"
  >
    We are getting you a quote
    <br />
    Please wait...
  </p>
</div>
`;
